qiita_plugins/target_gene/tgp/pick_otus.py

```diff
@@ -0,0 +1,140 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from os.path import join
+from functools import partial
+from glob import glob
+
+import requests
+
+from tgp.util import (update_job_step, execute_request_retry, system_call,
+                      format_payload)
+
+
+def write_parameters_file(fp, parameters):
+    """Write the QIIME parameters file
+
+    Parameters
+    ----------
+    fp : str
+        The paramters file filepath
+    parameters : dict
+        The commands parameters, keyed by parameter name
+    """
+    params = ['sortmerna_max_pos', 'similarity', 'sortmerna_coverage',
+              'threads']
+    with open(fp, 'w') as f:
+        for p in params:
+            f.write("pick_otus:%s\t%s" % (p, parameters[p]))
+
+
+def generate_pick_closed_reference_otus_cmd(filepaths, out_dir, parameters,
+                                            reference_fps):
+    """Generates the pick_closed_reference_otus.py command
+
+    Parameters
+    ----------
+    filepaths : list of (str, str)
+        The artifact's filepaths and their types
+    out_dir : str
+        The job output directory
+    parameters : dict
+        The command's parameters, keyed by parameter name
+    reference_fps : list of (str, str)
+        The reference filepaths and their types
+
+    Returns
+    -------
+    str
+        The pick_closed_reference_otus.py command
+    """
+    seqs_fp = None
+    for fp, fp_type in filepaths:
+        if fp_type == 'preprocessed_fasta':
+            seqs_fp = fp
+            break
+
+    if not seqs_fp:
+        raise ValueError("No sequence file found on the artifact")
+
+    output_dir = join(out_dir, 'cr_otus')
+    param_fp = join(out_dir, 'cr_params.txt')
+
+    write_parameters_file(param_fp, parameters)
+
+    reference_fp = None
+    taxonomy_fp = None
+    for rfp, rfp_type in reference_fps:
+        if rfp_type == 'reference_seqs':
+            reference_fp = rfp
+        elif rfp_type == 'reference_tax':
+            taxonomy_fp = rfp
+
```

Comment:
Nope, it is possible that it has `reference_tree`, but I'm not using it here, so not asking for it.